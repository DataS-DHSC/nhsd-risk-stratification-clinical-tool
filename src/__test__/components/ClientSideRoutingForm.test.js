import React from 'react';
import { render, fireEvent, createEvent } from '@testing-library/react';
import { useRouter } from 'next/router';
import ClientSideRoutingSupportForm from '../../components/ClientSideRoutingForm';

jest.mock('next/router');

const defaultProps = {
  children: <div>test</div>,
  method: 'POST',
  action: 'test/url',
  id: 'test',
};

describe('ClientSideRoutingForm', () => {
  afterEach(jest.clearAllMocks);

  it('should match snapshot', () => {
    const container = render(
      <ClientSideRoutingSupportForm {...defaultProps} />
    );

    expect(container.asFragment()).toMatchSnapshot();
  });

  it('should pass the correct url to the router when client side routing is enabled', () => {
    const router = { push: jest.fn() };
    useRouter.mockImplementationOnce(() => router);

    const props = {
      ...defaultProps,
    };

    const { container } = render(<ClientSideRoutingSupportForm {...props} />);
    const form = container.firstChild;
    const submitEvent = createEvent.submit(form);
    fireEvent(form, submitEvent);

    expect(submitEvent.defaultPrevented).toBe(true);
    expect(router.push).toHaveBeenCalledWith(props.action);
  });
});
