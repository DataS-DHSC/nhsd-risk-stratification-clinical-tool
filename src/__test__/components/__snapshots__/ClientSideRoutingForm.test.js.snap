// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientSideRoutingForm should match snapshot 1`] = `
<DocumentFragment>
  <form
    action="test/url"
    id="test"
    method="POST"
  >
    <div>
      test
    </div>
  </form>
</DocumentFragment>
`;
